File: WLIOTAndroidApplication/libwliotproxy/src/main/java/com/alterozoom/libwliotproxy/WLIOTProtocolDefs.java
package com.alterozoom.libwliotproxy;

public class WLIOTProtocolDefs
{
	public final static ByteArray funcCallMsg=ByteArray.fromString("call");
	public final static ByteArray funcAnswerOkMsg=ByteArray.fromString("ok");
	public final static ByteArray funcAnswerErrMsg=ByteArray.fromString("err");
	public final static ByteArray devSyncMsg=ByteArray.fromString("sync");
	public final static ByteArray devSyncrMsg=ByteArray.fromString("syncr");
	public final static ByteArray infoMsg=ByteArray.fromString("info");
	public final static ByteArray measurementMsg=ByteArray.fromString("meas");
	public final static ByteArray measurementBMsg=ByteArray.fromString("measb");
	public final static ByteArray identifyMsg=ByteArray.fromString("identify");
	public final static ByteArray identifyHubMsg=ByteArray.fromString("identify_hub");
	public final static ByteArray deviceInfoMsg=ByteArray.fromString("deviceinfo");
	public final static ByteArray stateChangedMsg=ByteArray.fromString("statechanged");
	public final static ByteArray hubMsg=ByteArray.fromString("#hub");
	public final static ByteArray hubDeviceIdentifiedMsg=ByteArray.fromString("device_identified");
	public final static ByteArray hubDeviceLostMsg=ByteArray.fromString("device_lost");
//	public final static QString queryVersionMsg;
//	public final static QString versionMsg;
	public final static byte msgDelim=10;
	public final static byte argDelim=124;
	public final static int identifyWaitTime=5000;//msecs
	public final static int syncWaitTime=5000;//msecs
	public final static ByteArray serverReadyMsg=ByteArray.fromString("server_ready");

	//reserved commands
	public final static ByteArray getSensorsCommand=ByteArray.fromString("#sensors");
	public final static ByteArray getControlsCommand=ByteArray.fromString("#controls");
	public final static ByteArray getStateCommand=ByteArray.fromString("#state");

	public final static int netDevicePort=4081;
	public final static int netDeviceSslPort=4082;
}

File: wliotproxy-src-base/libwliotproxy-base/src/WLIOTProtocolDefs.cpp
/*******************************************
Copyright 2017 OOO "LMS"

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

	http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.*/

#include "wliot/WLIOTProtocolDefs.h"

using namespace WLIOT;

const QByteArray WLIOTProtocolDefs::funcCallMsg="call";
const QByteArray WLIOTProtocolDefs::funcAnswerErrMsg="err";
const QByteArray WLIOTProtocolDefs::funcAnswerOkMsg="ok";
const QByteArray WLIOTProtocolDefs::devSyncMsg="sync";
const QByteArray WLIOTProtocolDefs::devSyncrMsg="syncr";
const QByteArray WLIOTProtocolDefs::measurementMsg="meas";
const QByteArray WLIOTProtocolDefs::measurementBMsg="measb";
const QByteArray WLIOTProtocolDefs::infoMsg="info";
const QByteArray WLIOTProtocolDefs::identifyMsg="identify";
const QByteArray WLIOTProtocolDefs::deviceInfoMsg="deviceinfo";
const QByteArray WLIOTProtocolDefs::stateChangedMsg="statechanged";
const QByteArray WLIOTProtocolDefs::hubMsg="#hub";
const QByteArray WLIOTProtocolDefs::hubDeviceIdentifiedMsg="device_identified";
const QByteArray WLIOTProtocolDefs::hubDeviceLostMsg="device_lost";
const QByteArray WLIOTProtocolDefs::serverReadyMsg="server_ready";
//const QString WLIOTConfig::queryVersionMsg=QString("queryversion");
//const QString WLIOTConfig::versionMsg=QString("version");
const char WLIOTProtocolDefs::msgDelim=10;//\n
const char WLIOTProtocolDefs::argDelim=124;//|
const int WLIOTProtocolDefs::identifyWaitTime=5000;
const int WLIOTProtocolDefs::syncWaitTime=3000;

const QByteArray WLIOTProtocolDefs::getSensorsCommand="#sensors";
const QByteArray WLIOTProtocolDefs::getControlsCommand="#controls";
const QByteArray WLIOTProtocolDefs::getStateCommand="#state";
const QByteArray WLIOTProtocolDefs::identifyHubCommand="#identify_hub";

const quint16 WLIOTProtocolDefs::netDevicePort=4081;
const quint16 WLIOTProtocolDefs::netDeviceSslPort=4082;
